tartiflette/schema/schema.py

```diff
@@ -582,8 +631,217 @@ def _validate_directive_implementation(self) -> List[str]:
                 attr = getattr(directive.implementation, expected, None)
                 if attr and not is_valid_coroutine(attr):
                     errors.append(
-                        f"Directive {directive.name} Method {expected} is not awaitable"
+                        f"Directive {directive.name} Method "
+                        f"{expected} is not awaitable"
+                    )
+        return errors
+
+    def _validate_enum_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLEnumTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLEnumType, "ENUM"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                values = [x.name for x in extended.values]
+                for value in extension.values:
+                    if value.name in values:
+                        errors.append(
```

Comment:
What do you think about writing a little function which would returns a list of errors when a value is already defined? Since this treatment is performed a lot it could worth it to refactor it and also possibly use `set` instead for nested `for` loops:
```python
def already_defined(
    reference_values: List[str],
    extended_values: List[str],
    extended_name: str,
    message: str,
) -> List[str]:
    return [
        message.format(
            existing_value=existing_value,
            extended_name=extended_name
        )
        for existing_value in set(reference_values) & set(extended_values)
    ]
```

Which, in this case, could be use like:
```python
errors.extend(
    already_defined(
        [x.name for x in extended.values],
        extension.values,
        extended.name,
        "Can't add < {existing_value} > value to < {extended_name} > ENUM, cause value already exists.",
    )
)
```

Reply:
Yes, good idea, i was also thinking that all these "validation" method looked to much alike. I can also factor it a little bit, but you're also in the process of re-writing the validation for the SDL part, I didn't knew if it was worth it to pass a lot of time to factorize this code that going to be redone soon. If you think it is worth it, i'll do it.